tests/datasetops/test_caching.py

```diff
@@ -0,0 +1,330 @@
+from datasetops.dataset import Dataset
+from typing import List
+from datasetops.cache import Cache
+from datasetops import loaders
+import numpy as np
+from testing_utils import get_test_dataset_path, \
+    from_dummy_numpy_data, DATASET_PATHS  # type:ignore
+import dill
+
+
+def test_cachable():
+    path = get_test_dataset_path(DATASET_PATHS.FOLDER_DATASET_GROUP_DATA)
+    test, _ = loaders.from_folder_dataset_group_data(path)
+
+    def read_text(path):
```

Comment:
We could consider adding automatic readers for txt and bin data as part of the library

Reply:
Yes, we can do this. While doing it, it would be good to add tests, so they will use both internal and user-defined readers